test/js/node/http/node-http.test.ts

```diff
@@ -158,6 +158,23 @@ describe("node:http", () => {
       server.close();
     });
 
+    it("should use the provided port", async () => {
+      const server = http.createServer(() => {});
+      server.listen(3000);
+      const { port } = server.address();
+      expect(port).toEqual(3000);
+      server.close();
+    });
+
+    it("should assign a random port when undefined", async () => {
+      const server = http.createServer(() => {});
+      server.listen(undefined);
+      const { port } = server.address();
+      expect(port).not.toBe(3000);
```

Comment:
This can be flaky, because a random port could end up being `3000`! Instead, a way to check this is by creating 2 servers, and validating that the ports are not the same.

Reply:
Yeah, makes sense, thanks!